garage/tf/algos/npo.py

```diff
@@ -244,6 +241,9 @@ def _build_policy_loss(self, i):
             adv_valid = filter_valids(
                 adv_flat, i.flat.valid_var, name="adv_valid")
 
+            if self.policy.recurrent:
+                advantages = tf.reshape(advantages, [-1, self.max_path_length])
```

Comment:
hmm what shape do the recurrent policies need?

`compute_advantages` should output shape `(trajectory, step, advantage)` and then `adv_flat` should have shape `(trajectory x time, advantage)`.

Reading this, it looks like your recurrent policy is getting shape `(trajectory x advantage, step)`, which I don't really understand.

Reply:
`compute_advantages` is actually output shape `(trajectory x step x advantage)` because here. https://github.com/rlworkgroup/garage/blob/074c2aa100617c27f73d34a31ab257ccedaf8feb/garage/tf/misc/tensor_utils.py#L218

Reply:
I think that's what I said above?

What is the shape of advantages after you reshape it here?

Reply:
Sorry, I mean `(trajectory x step x advantage, )`.

After reshaping, the shape of advantages is `(trajectory x advantage, step). Let me check if the shape is correct for recurrent policy.